### PR TITLE
fix: unexpected panic when allotment sum is not 1

### PR DESCRIPTION
**Pull request by ascandone**
Fixes https://github.com/formancehq/stack/issues/1473

### CODE REVIEW

**Review by gfyrag — APPROVED** [review]

**Comment by gfyrag** [comment]
@ascandone Thks for the PR! Can you rebase please?

**Review by paul-nicolas — APPROVED** [review]

**Comment by ascandone** [comment]
> @ascandone Thks for the PR! Can you rebase please?

Hey @gfyrag , just a quick ping, the workflow is waiting for approval to be run, so the merge queue has been stuck

**Pull request merged**
